src/lib.rs

```diff
@@ -298,10 +298,14 @@ broker_handler!(AuthHandler, |app, req| {
     // Per the OAuth2 spec, we may redirect to the RP once we have validated client_id and
     // redirect_uri. In our case, this means we make redirect_uri available to error handling.
     let client_id = try_get_param!(params, "client_id");
-    let redirect_uri = try!(
-        Url::parse(try_get_param!(params, "redirect_uri"))
-            .map_err(|_| BrokerError::Input("redirect_uri is not a valid URL".to_string()))
-    );
+    let raw_redirect_uri = try_get_param!(params, "redirect_uri");
+    let redirect_uri = try!(Url::parse(raw_redirect_uri)
+            .map_err(|_| BrokerError::Input("redirect_uri is not a valid URL".to_string())));
+    if app.allowed_redirect_uris.len() > 0 {
```

Comment:
nit: `!app.allowed_redirect_uris.is_empty()` instead of `app.allowed_redirect_uris.len() > 0`?